src/ui/BindingBase.hh

```diff
@@ -0,0 +1,91 @@
+#ifndef RA_UI_BINDINGBASE_H
+#define RA_UI_BINDINGBASE_H
+#pragma once
+
+#include "ui/ViewModelBase.hh"
+
+namespace ra {
+namespace ui {
+
+class BindingBase : protected ViewModelBase::NotifyTarget
```

Comment:
`BindingBase` attaches to a `ViewModelBase` through the `NotifyTarget` callback. It also has methods for reading data from the `ViewModelBase` generically using a `ModelProperty`